src/components/invoices/InvoiceSelectionContainer.js

```diff
@@ -1,17 +1,16 @@
-import React, { Component } from "react";
-import PropTypes from "prop-types";
-import { withStyles } from "@material-ui/core/styles";
-import { withTranslation } from "react-i18next";
-import Paper from "@material-ui/core/Paper";
-import LinearProgress from "@material-ui/core/LinearProgress";
-import Typography from "@material-ui/core/Typography";
-import FormControl from "@material-ui/core/FormControl";
+import React, { useState, useEffect } from "react";
+import { useTranslation } from "react-i18next";
+import { FormControl, LinearProgress, Paper, Typography } from "@material-ui/core";
+
+import { makeStyles } from "@material-ui/styles";
 import OrgUnitAutoComplete from "./OrgUnitAutoComplete";
-import PeriodPicker from "../shared/PeriodPicker";
 import OuPicker from "./OuPicker";
+
+import PeriodPicker from "../shared/PeriodPicker";
+import searchOrgunit from "./searchOrgunit";
 import SelectionResultsContainer from "./SelectionResultsContainer";
-import PluginRegistry from "../core/PluginRegistry";
-import debounce from "lodash/debounce";
+
+import useDebounce from "../dataentry/useDebounce";
```

Comment:
Here we are a bit breaking the wall between modules.
I would move this useDebounce in the shared folder

Reply:
got it! see https://github.com/BLSQ/blsq-report-components/commit/2ae0bfe7ca8e301926a83f8a64fb3ac41072ff52